src/test/java/com/microsoft/sqlserver/jdbc/exception/ExceptionTest.java

```diff
@@ -0,0 +1,123 @@
+     * 
+     * @throws Exception
+     */
+    @Test
+    public void testBulkCSVFileRecordExceptionCause() throws Exception {
+        String filePath = getCurrentClassPath();
+
+        try {
+            SQLServerBulkCSVFileRecord scvFileRecord = new SQLServerBulkCSVFileRecord(filePath + inputFile, "invalid_encoding", true);
+        }
+        catch (Exception e) {
+            if (!(e instanceof SQLServerException)) {
+                throw e;
+            }
+
+            assertTrue(null != e.getCause(), "Cause should not be null.");
+            assertTrue(e.getCause() instanceof UnsupportedEncodingException, "Cause should be instance of UnsupportedEncodingException.");
+        }
+    }
+
+    /**
+     * 
+     * @return location of resource file
+     */
+    static String getCurrentClassPath() {
+
+        try {
+            String className = new Object() {
+            }.getClass().getEnclosingClass().getName();
+            String location = Class.forName(className).getProtectionDomain().getCodeSource().getLocation().getPath() + "/";
+            URI uri = new URI(location.toString());
+            return uri.getPath();
+        }
+        catch (Exception e) {
+            fail("Failed to get CSV file path. " + e.getMessage());
+        }
+        return null;
+    }
+
+    String waitForDelaySPName = "waitForDelaySP";
+    final int waitForDelaySeconds = 10;
+
+    /**
+     * Test the SQLServerException has the proper cause when socket timeout occurs.
+     * 
+     * @throws Exception
+     * 
+     */
+    @Test
+    public void testSocketTimeoutExceptionCause() throws Exception {
+        SQLServerConnection conn = null;
+        try {
+            conn = (SQLServerConnection) DriverManager.getConnection(connectionString);
+
+            dropWaitForDelayProcedure(conn);
+            createWaitForDelayPreocedure(conn);
+
+            conn = (SQLServerConnection) DriverManager.getConnection(connectionString + ";socketTimeout=" + (waitForDelaySeconds * 1000 / 2) + ";");
+
+            try {
+                conn.createStatement().execute("exec " + waitForDelaySPName);
+                throw new Exception("Exception for socketTimeout is not thrown.");
+            }
+            catch (Exception e) {
+                if (!(e instanceof SQLServerException)) {
+                    throw e;
+                }
+
+                assertTrue(null != e.getCause(), "Cause should not be null.");
+                assertTrue(e.getCause() instanceof SocketTimeoutException, "Cause should be instance of SocketTimeoutException.");
+            }
+        }
+        finally {
+            if (null != conn) {
+                conn.close();
+            }
+        }
+    }
+
+    private void dropWaitForDelayProcedure(SQLServerConnection conn) throws SQLException {
+        String sql = " IF EXISTS (select * from sysobjects where id = object_id(N'" + waitForDelaySPName
```

Comment:
This PR is pretty old, before we have this `Utils.dropObjectIfExists(...)` method.